### PR TITLE
kak-lsp: new recipe

### PR DESCRIPTION
**Pull request by t6**

### CODE REVIEW

File: dev-util/kak_lsp/kak_lsp-12.1.0.recipe [hunk 1]
```diff
@@ -0,0 +1,449 @@
+SUMMARY="Kakoune Language Server Protocol Client"
+DESCRIPTION="\
+Kak-lsp is a Language Server Protocol client for Kakoune implemented
+in Rust.
```
Comment (Begasus) [inline dev-util/kak_lsp/kak_lsp-12.1.0.recipe:4]: DESCRIPTION starts on the same line, lines can go up to 100 characters now (this has been changed recently): https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#general-tips

**Comment by Begasus** [comment]
This should go to `app-editors` instead of `dev-util`, see: https://gpo.zugaina.org/Overlays/guru/app-editors/kak-lsp

File: app-editors/kak_lsp/kak_lsp-12.1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	kak_lsp$secondaryArchSuffix = $portVersion
+	cmd:kak_lsp
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	kakoune$secondaryArchSuffix
```
Comment (Begasus) [inline app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: It's the other way around, kakoune can have an optional dependency for kak-lsp, no need for this here ;)
```diff
- 	kakoune$secondaryArchSuffix
```

Comment (t6) [reply app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: kak-lsp is useless without kakoune. On Arch kakoune optionally depends on kak-lsp and kak-lsp depends on kakoune.

Comment (Begasus) [reply app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: My bad there, didn't check on this one ;) So let's leave it in (build in progress here atm)

Comment (t6) [reply app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: How did the build go?

Comment (korli) [reply app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: if you want kak-lsp to be proposed to install along kakoune:
`SUPPLEMENTS="kakoune$secondaryArchSuffix"`

Comment (Begasus) [reply app-editors/kak_lsp/kak_lsp-12.1.0.recipe:393]: Build went OK, now I remember why I don't like rust packages to build :)

**Comment by Begasus** [comment]
Merging this one, the other is already merged, keep an eye out on the buildmaster, thanks!

**Pull request merged**